src/EditorFeatures/Core/ExternalAccess/VSTypeScript/VSTypeScriptInProcLanguageClient.cs

```diff
@@ -57,6 +58,21 @@ public override ServerCapabilities GetCapabilities(ClientCapabilities clientCapa
         serverCapabilities.ProjectContextProvider = true;
 
         serverCapabilities.SupportsDiagnosticRequests = true;
+        serverCapabilities.DiagnosticProvider = new()
+        {
+            SupportsMultipleContextsDiagnostics = true,
+            DiagnosticKinds =
+            [
+                new(PullDiagnosticCategories.Task),
+                new(PullDiagnosticCategories.EditAndContinue),
+                new(PullDiagnosticCategories.WorkspaceDocumentsAndProject),
+                new(PullDiagnosticCategories.DocumentCompilerSyntax),
+                new(PullDiagnosticCategories.DocumentCompilerSemantic),
+                new(PullDiagnosticCategories.DocumentAnalyzerSyntax),
+                new(PullDiagnosticCategories.DocumentAnalyzerSemantic),
```

Comment:
can you try just doing WorspaceDocsProj, CompilerSyntax and CompilerSemantic.  Given that our api into you is just:

```
    Task<ImmutableArray<Diagnostic>> AnalyzeProjectAsync(Project project, CancellationToken cancellationToken);
    Task<ImmutableArray<Diagnostic>> AnalyzeDocumentSyntaxAsync(Document document, CancellationToken cancellationToken);
    Task<ImmutableArray<Diagnostic>> AnalyzeDocumentSemanticsAsync(Document document, CancellationToken cancellationToken);
```

i think that should be all that's necessary.

Reply:
Hmmm I tried just these and non-TODO diagnostics are gone:
```cs
serverCapabilities.DiagnosticProvider = new()
{
    SupportsMultipleContextsDiagnostics = true,
    DiagnosticKinds =
    [
        new(PullDiagnosticCategories.Task),
        new(PullDiagnosticCategories.WorkspaceDocumentsAndProject),
        new(PullDiagnosticCategories.DocumentCompilerSyntax),
        new(PullDiagnosticCategories.DocumentCompilerSemantic),
    ]
};
```

Reply:
Never mind, it works if I keep the analyzer kinds instead. @CyrusNajmabadi let me know if this is what you were suggesting :)

Reply:
Definitely odd.  But that's ok with me.  I guess we treat all of ts as an analyzer, not a compiler.  Are you able to remove the compiler ones?